addons/editor-theme3/theme3.js

```diff
@@ -260,6 +272,13 @@ export default async function ({ addon, console }) {
     return oldFieldMatrixCreateButton.call(this, fill);
   };
 
+  const oldFieldVerticalSeparatorInit = Blockly.FieldVerticalSeparator.prototype.init;
+  Blockly.FieldVerticalSeparator.prototype.init = function () {
+    // Vertical line between extension icon and block label
+    oldFieldVerticalSeparatorInit.call(this);
+    if (textMode === "black") this.lineElement_.setAttribute("stroke", this.sourceBlock_.getColourTertiary());
+  };
+
```

Comment:
I think you forgot dynamicDisable support here.

dynamicDisable is coded weirdly on this addon. The `addon.self.disabled` checks aren't found where you expect in other traps.